Changes

```diff
@@ -247,6 +247,9 @@ Working version
   (Ulysse Gérard, Leo White, review by Antonin Décimo, Gabriel Scherer,
    Samuel Vivien, Florian Angeletti and Jacques Garrigue)
 
+- #13744: Refactor in `collect_apply_args`
+  (Samuel Vivien, review by ...)
```

Comment:
The reviewers need to be named and the PR rebased, otherwise looks ready to merge.

Reply:
It is done